src/rf/js/src/user/components.js

```diff
@@ -0,0 +1,381 @@
+        this.handleServerFailure(response);
+    },
+
+    render: function() {
+        var content = (
+            <div>
+                {this.renderErrors()}
+                <div className="link-block">
+                    <a href="/forgot/" data-url="/forgot/"
+                        className="text-muted small forgot">Forgot?</a>
+                    <a href="/send-activation/" data-url="/send-activation/"
+                        className="text-muted small send-activation">Send Activation Email</a>
+                    <a href="/sign-up/" data-url="/sign-up/"
+                        className="text-muted small sign-up">Sign Up</a>
+                </div>
+                <form method="post" onSubmit={this.validate}>
+                    <FormGroup name="username" displayName="Username"
+                        ref="username"
+                        value={this.getModel().get('username')}/>
+                    <FormGroup name="password" displayName="Password"
+                        ref="password" type="password"
+                        value={this.getModel().get('password')}/>
+                    <div className="form-action">
+                        <input type="submit" className="btn btn-secondary btn-block" value="Log In" />
+                    </div>
+                </form>
+            </div>
+        );
+
+        return <UserScreen content={content} />;
+    }
+});
+
+var SignUpScreen = React.createBackboneClass({
+    mixins: [ModelValidationMixin],
+
+    onValidateFailure: function() {
+        this.refs.username.setFocus();
+    },
+
+    getFormData: function() {
+        return {
+            username: this.refs.username.getValue(),
+            email: this.refs.email.getValue(),
+            password1: this.refs.password1.getValue(),
+            password2: this.refs.password2.getValue(),
+            agreed: this.refs.agreed.getValue()
+        };
+    },
+
+    setFields: function() {
+        this.getModel().set(this.getFormData(), { silent: true });
+    },
+
+    primaryAction: function() {
+        this.getModel()
+            .fetch({
+                method: 'POST',
+                data: this.getFormData()
+            })
+            .done(_.bind(this.handleServerSuccess, this))
+            .fail(_.bind(this.handleServerFailure, this));
+    },
+
+    handleSuccess: function() {
+    },
+
+    handleFailure: function(response) {
+        this.handleServerFailure(response);
+    },
+
+    render: function() {
+        var content = 'Please check your email to activate your account.';
+        if (!this.getModel().get('success')) {
+            content = (
+                <form method="post" onSubmit={this.validate}>
+                    {this.renderErrors()}
+                    <FormGroup name="username" displayName="Username" ref="username" value={this.getModel().get('username')}/>
+                    <FormGroup name="email" displayName="Email" ref="email" value={this.getModel().get('email')}/>
```

Comment:
I still think this should be of type "email"

If you don't want built in browser validation you can add `noValidate` at line 238